xc/readln.ml

```diff
@@ -0,0 +1,57 @@
+(* Read newline-delimited strings from a file descriptor *)
+
+type result      = Ok of string list | Error of string | EOF
+
+let input        = Hashtbl.create 100 (* holds unconsumed input per fd *)
+let msgs         = Hashtbl.create 100 (* holds complete strings per fd *)
```

Comment:
Why is `msgs` needed? Surely complete messages are just returned and don't need to be stored anywhere (except perhaps a temporary list in the `read` function?

Reply:
Good point. They are all returned.